app/uk/gov/hmrc/helptosave/services/UserCapService.scala

```diff
@@ -24,23 +24,22 @@ import uk.gov.hmrc.helptosave.config.AppConfig
 import uk.gov.hmrc.helptosave.models.UserCapResponse
 import uk.gov.hmrc.helptosave.repo.UserCapStore
 import uk.gov.hmrc.helptosave.repo.UserCapStore.UserCap
-import uk.gov.hmrc.helptosave.util._
+import uk.gov.hmrc.helptosave.util.{WithMdcExecutionContext, _}
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.control.NonFatal
 
 @ImplementedBy(classOf[UserCapServiceImpl])
 trait UserCapService {
 
-  def isAccountCreateAllowed(): Future[UserCapResponse]
+  def isAccountCreateAllowed()(implicit ec: ExecutionContext): Future[UserCapResponse]
 
-  def update(): Future[Unit]
+  def update()(implicit ec: ExecutionContext): Future[Unit]
 
 }
 
 @Singleton
-class UserCapServiceImpl @Inject() (userCapStore: UserCapStore)(implicit appConfig: AppConfig) extends UserCapService with Logging {
+class UserCapServiceImpl @Inject() (userCapStore: UserCapStore)(implicit appConfig: AppConfig) extends UserCapService with Logging with WithMdcExecutionContext {
```

Comment:
do we need to extend with `WithMdcExecutionContext` here?

Reply:
I think I again missed pushing the commit - let me check

Reply:
pushed the commit